File: src/main/java/com/example/probation/service/impl/RoleServiceImpl.java
package com.example.probation.service.impl;

import com.example.probation.core.entity.Role;
import com.example.probation.repository.RoleRepository;
import com.example.probation.service.RoleService;
import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

@Service
@Transactional
@RequiredArgsConstructor
public class RoleServiceImpl implements RoleService {
    public final RoleRepository roleRepository;

    @Override
    public Role getRoleByRoleName(final String roleName) {
        return roleRepository.getRoleByRoleName(roleName);
    }
}

File: src/main/kotlin/com/example/probation/service/impl/RoleServiceImpl.kt
package com.example.probation.service.impl

import com.example.probation.exception.EntityNotFoundException
import com.example.probation.repository.RoleRepository
import com.example.probation.service.RoleService
import org.springframework.stereotype.Service

@Service
class RoleServiceImpl(private val roleRepository: RoleRepository) : RoleService {

    override fun getRoleByRoleName(roleName: String) =
        roleRepository.getRoleByRoleName(roleName) ?: throw EntityNotFoundException("{entity.notfound}")
}
